google/cloud/speech/client.py

```diff
@@ -139,44 +192,115 @@ def sync_recognize(self, content, source_uri, encoding, sample_rate,
                     between 0 and 1.
         """
 
-        if content is None and source_uri is None:
-            raise ValueError('content and source_uri cannot be both '
-                             'equal to None')
-
-        if content is not None and source_uri is not None:
-            raise ValueError('content and source_uri cannot be both '
-                             'different from None')
+        data = _build_request_data(content, source_uri, encoding,
+                                   sample_rate, language_code,
+                                   max_alternatives, profanity_filter,
+                                   speech_context)
 
-        if encoding is None:
-            raise ValueError('encoding cannot be None')
-        if sample_rate is None:
-            raise ValueError('sample_rate cannot be None')
+        api_response = self.connection.api_request(
+            method='POST', path='speech:syncrecognize', data=data)
 
-        if content is not None:
-            audio = {'content': b64encode(_to_bytes(content))}
+        if len(api_response['results']) == 1:
+            return api_response['results'][0]['alternatives']
         else:
-            audio = {'uri': source_uri}
+            raise ValueError('result in api should have length 1')
 
-        config = {'encoding': encoding, 'sampleRate': sample_rate}
 
-        if language_code is not None:
-            config['languageCode'] = language_code
-        if max_alternatives is not None:
-            config['maxAlternatives'] = max_alternatives
-        if profanity_filter is not None:
-            config['profanityFilter'] = profanity_filter
-        if speech_context is not None:
-            config['speechContext'] = {'phrases': speech_context}
+def _build_request_data(content, source_uri, encoding, sample_rate,
```

Comment:
This method signature is repeated three times. Once here, once in `async_recognize` and once again in `sync_recognize`. If anyone has suggestions on how to DRY this up I'm happy to implement.